File: mkdocs/snippets/devbook/accounts/create-from-mnemonic.py
import os

from symbolchain.Bip32 import Bip32
from symbolchain.facade.SymbolFacade import SymbolFacade

# Initialize the facade for the testnet network [>step-1]
facade = SymbolFacade('testnet')
# [<step-1]
# Use an existing mnemonic if provided, otherwise generate a random one [>step-2]
bip32 = Bip32()
mnemonic = os.getenv('MNEMONIC')
if mnemonic:
	print("Loading mnemonic phrase from environment variable...")
else:
	print("Generating random mnemonic phrase...")
	mnemonic = bip32.random()
print(f'Mnemonic phrase: {mnemonic}')
# [<step-2]
# Load password from environment variable or use default [>step-3]
password = os.getenv('PASSWORD', 'correcthorsebatterystaple')
print(f'Password: {password}')

# Derive a root Bip32 node from the mnemonic and a password
root_node = bip32.from_mnemonic(mnemonic, password)
# [<step-3]
# Derive a child Bip32 node for the account at index 0 [>step-4]
account_index = 0
child_node = root_node.derive_path(facade.bip32_path(account_index))
# [<step-4]
# Convert the Bip32 node to a signing key pair [>step-5]
key_pair = facade.bip32_node_to_key_pair(child_node)

# Derive the address from the public key
address = facade.network.public_key_to_address(key_pair.public_key)

# Output the account details
print(f'Address: {address}')
print(f'Public key: {key_pair.public_key}')
print(f'Private key: {key_pair.private_key}') # [<step-5]

File: mkdocs/snippets/devbook/accounts/create_from_mnemonic.java
//JAVA 21+
//DEPS org.symbol:symbol-sdk:3.3.1

import org.symbol.sdk.Bip32;
import org.symbol.sdk.CryptoTypes;
import org.symbol.sdk.facade.SymbolFacade;
import org.symbol.sdk.symbol.Address;
import org.symbol.sdk.symbol.KeyPair;

final class CreateFromMnemonic {
	private CreateFromMnemonic() {
	}

	public static void main(final String[] args) {
		// Initialize the facade for the testnet network [>step-1]
		final SymbolFacade facade = new SymbolFacade("testnet");
		// [<step-1]
		// Use an existing mnemonic if provided, [>step-2]
		// otherwise generate a random one.
		final Bip32 bip32 = new Bip32(SymbolFacade.BIP32_CURVE_NAME);
		String mnemonic = System.getenv("MNEMONIC");
		if (null != mnemonic) {
			System.out.println(
				"Loading mnemonic phrase from environment variable...");
		} else {
			System.out.println("Generating random mnemonic phrase...");
			mnemonic = bip32.random();
		}
		System.out.println("Mnemonic phrase: " + mnemonic);
		// [<step-2]
		// Load password from environment variable or use default [>step-3]
		final String password = System.getenv().getOrDefault(
			"PASSWORD", "correcthorsebatterystaple");
		System.out.println("Password: " + password);

		// Derive a root Bip32 node from the mnemonic and a password
		final Bip32.Bip32Node rootNode = bip32.fromMnemonic(
			mnemonic, password);
		// [<step-3]
		// Derive a child Bip32 node for the account at index 0 [>step-4]
		final int accountIndex = 0;
		final Bip32.Bip32Node childNode = rootNode.derivePath(
			facade.bip32Path(accountIndex));
		// [<step-4]
		// Convert the Bip32 node to a signing key pair [>step-5]
		final KeyPair keyPair = SymbolFacade.bip32NodeToKeyPair(childNode);

		// Derive the address from the public key
		final CryptoTypes.PublicKey publicKey = keyPair.getPublicKey();
		final Address address = facade.network.publicKeyToAddress(
			publicKey);

		// Output the account details
		System.out.println("Address: " + address);
		System.out.println("Public key: " + publicKey);
		System.out.println(
			"Private key: " + keyPair.getPrivateKey()); // [<step-5]
	}
}
